### DISCUSSION TITLE
Listen for a click event without blocking the elements below it

### DISCUSSION THREAD

**Discussion by MarcusChamberlain** [AvaloniaUI/Avalonia#10848] [Q&A]
I'm trying to intercept a click event to see if the user is clicking over a certain panel, but I don't want to block the elements below it (it's so I can check which of my panels is active and bring its z-index forward). But I'm not sure how to do this. I've been reading about routed events, but I can't connect the dots on how to use them, so I'm not sure if this is the use case for them.

![Screenshot 2023-03-30 001222](https://user-images.githubusercontent.com/12500226/228727563-477e28f1-00af-4109-84f5-1641be773f50.png)

I want to click the pink rectangle and register a click event, but I don't want to block any interactions (including pointer over) from the button underneath. Basically the button should continue to work like there's no overlay, but the pink overlay can passively listen in and log click events. 

Naively I thought if I just added an "OnPointerPressed" event handler in the code behind and set the "e.Handled = false;" that would do it, but alas it's not that simple.

Any advice on how to achieve this?

**Comment by timunie** [comment]
I wonder if an adorner could could help in this situation? https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/AdornerLayerPage.xaml

**Reply by timunie** [reply]
@MarcusChamberlain also a description what the pink border should exactly do when the user clicks would help a lot to get some ideas.
